rust/frameworks/tw_utxo/src/transaction/standard_transaction/mod.rs

```diff
@@ -0,0 +1,356 @@
+    /// | >= 500000000   | UNIX timestamp at which this transaction is unlocked
+    ///
+    /// If all inputs have final (`0xffffffff`) sequence numbers then `lockTime` is irrelevant.
+    /// Otherwise, the transaction may not be added to a block until after `lockTime`.
+    pub locktime: u32,
+}
+
+impl TransactionInterface for Transaction {
+    type Input = TransactionInput;
+    type Output = TransactionOutput;
+
+    fn version(&self) -> i32 {
+        self.version
+    }
+
+    fn inputs(&self) -> &[Self::Input] {
+        &self.inputs
+    }
+
+    fn inputs_mut(&mut self) -> &mut [Self::Input] {
+        &mut self.inputs
+    }
+
+    fn replace_inputs(&mut self, inputs: Vec<Self::Input>) {
+        self.inputs = inputs;
+    }
+
+    fn outputs(&self) -> &[Self::Output] {
+        &self.outputs
+    }
+
+    fn outputs_mut(&mut self) -> &mut [Self::Output] {
+        &mut self.outputs
+    }
+
+    fn change_amount(&self) -> Option<Amount> {
+        Some(self.outputs.last()?.value)
+    }
+
+    fn set_change_amount(&mut self, change: Amount) -> bool {
+        // TODO: Is it a good idea to just assume that the last output is the
+        // change output?
+        let Some(change_output) = self.outputs.last_mut() else {
+            return false;
+        };
+
+        change_output.value = change;
+
+        true
+    }
+
+    fn replace_outputs(&mut self, outputs: Vec<Self::Output>) {
+        self.outputs = outputs;
+    }
+
+    fn has_witness(&self) -> bool {
+        self.inputs.iter().any(|input| input.has_witness())
+    }
+
+    fn locktime(&self) -> u32 {
+        self.locktime
+    }
+}
+
+impl Transaction {
+    pub fn txid(&self) -> Vec<u8> {
+        // TODO
+        vec![]
+    }
+    /// Returns the same transaction with [`TransactionInput::script_witness`] being empty.
+    /// TODO: Why?
+    pub fn without_witness(&self) -> Transaction {
+        let mut without_witness = self.clone();
+        for input in without_witness.inputs.iter_mut() {
+            input.clear_witness();
+        }
+        without_witness
+    }
+    pub fn encode_out(&self) -> Vec<u8> {
+        let mut stream = Stream::new();
+        self.encode(&mut stream);
+        stream.out()
+    }
+    pub fn base_size(&self) -> usize {
+        let mut s = 0;
+        // Base transaction size.
+        s += VERSION_SIZE;
+        s += LOCKTIME_SIZE;
+
+        // Consider extended format in case witnesses are to be serialized.
+        if self.has_witness() {
+            s += WITNESS_FLAG_MARKER;
+        }
+
+        s += CompactInteger::from(self.inputs().len()).serialized_len();
+        s += CompactInteger::from(self.outputs().len()).serialized_len();
+        s
+    }
+    pub fn size(&self) -> usize {
+        let mut s = self.base_size();
+        self.inputs().iter().for_each(|input| s += input.size());
+        self.outputs().iter().for_each(|output| s += output.size());
+        s
+    }
```

Comment:
Do you think it's safer to just call `Transaction::encode` function? It can also simplify the work on Bitcoin forks later